src/Pomelo.EntityFrameworkCore.MySql/Extensions/MySqlDbContextOptionsExtensions.cs

```diff
@@ -87,5 +90,25 @@ private static MySqlOptionsExtension GetOrCreateExtension(DbContextOptionsBuilde
                 ? new MySqlOptionsExtension(existing)
                 : new MySqlOptionsExtension();
         }
+
+        private static bool IsAbleToUseDateTime6(string connectionStr)
```

Comment:
I've got some time to look into this today.  Let me take a shot at adding some tests around what you have and I'll push to this PR.